shared/common-adapters/input.desktop.js

```diff
@@ -1,60 +1,91 @@
 // @flow
 import React, {Component} from 'react'
-import {TextField} from 'material-ui'
+import {findDOMNode} from 'react-dom'
+import Box from './box'
+import Text, {getStyle as getTextStyle} from './text.desktop'
 import {globalStyles, globalColors} from '../styles'
-import {getStyle} from './text'
 
 import type {Props} from './input'
 
 type State = {
-  value: ?string,
+  value: string,
   focused: boolean,
 }
 
 class Input extends Component<void, Props, State> {
   state: State;
-  _textField: any;
+  _input: any;
 
   constructor (props: Props) {
     super(props)
 
     this.state = {
-      value: props.value,
+      value: props.value || '',
       focused: false,
     }
   }
 
-  getValue (): ?string {
-    return this.state.value
+  componentDidMount () {
+    this._autoResize()
+  }
+
+  componentWillReceiveProps (nextProps: Props) {
+    if (nextProps.hasOwnProperty('value')) {
+      this.setState({value: nextProps.value || ''})
+      this._autoResize()
+    }
+  }
+
+  getValue (): string {
+    return this.state.value || ''
   }
 
   setValue (value: string) {
-    this.setState({value})
+    this.setState({value: value || ''})
   }
 
   clearValue () {
-    this.onChange({target: {value: null}})
+    this._onChange({target: {value: ''}})
   }
 
-  onChange (event: {target: {value: ?string}}) {
-    this.setState({value: event.target.value})
-    this.props.onChange && this.props.onChange(event)
+  _onChange = (event: {target: {value: ?string}}) => {
+    this.setState({value: event.target.value || ''})
+    this._autoResize()
+
     this.props.onChangeText && this.props.onChangeText(event.target.value || '')
   }
 
+  _autoResize () {
+    if (!this.props.multiline) {
+      return
+    }
+
+    const node = this._inputNode()
+    if (!node) {
+      return
+    }
+
+    node.style.height = 'auto'
+    node.style.height = `${node.scrollHeight}px`
+  }
+
+  _inputNode () {
+    return findDOMNode(this._input)
+  }
+
   focus () {
-    this._textField && this._textField.focus()
+    this._input && this._inputNode().focus()
   }
 
   select () {
-    this._textField && this._textField.select()
+    this._input && this._inputNode().select()
   }
 
   blur () {
-    this._textField && this._textField.blur()
+    this._input && this._inputNode().blur()
   }
 
-  _onKeyDown (e: SyntheticKeyboardEvent) {
+  _onKeyDown = (e: SyntheticKeyboardEvent) => {
```

Comment:
Because this is an arrow function and i'm passing this into the render function below. This is actually the best pattern (i think) currently so you have a stored arrow that's not recomputed in render (this'll stop re-renders in the children)